app/ide-desktop/lib/client/src/index.ts

```diff
@@ -205,6 +225,41 @@ class App {
             })
         }
         electron.ipcMain.on(ipc.channel.quit, () => electron.app.quit())
+        // FIXME [NP]: Move this to the authentication package.
+        electron.ipcMain.on(ipc.channel.openExternalUrl, (event, url) => opener(url))
+        //electron.ipcMain.on(ipc.channel.setAuthenticatedRedirectCallback, (event, callback) => callback())
+    }
+
+    /**
+     * Initialize the listener for `open-url` events.
+     *
+     * This listener is used to open a page in *this* application window, when the user is
+     * redirected to a URL with a protocol supported by this application.
+     *
+     * For example, when the user completes an OAuth sign in flow (e.g., through Google), they are
+     * redirected to a URL like `enso://auth?code=...`. This listener will intercept that URL
+     * and open the page `auth?code=...` in the application window.
+     */
+    initOpenUrlListener() {
+        electron.app.on('open-url', (event, url) => {
+            // Prevent the default behavior (don't open `enso://auth?code=...` URL in the window).
+            event.preventDefault()
+
+            // FIXME [NP]: delete this log
+            console.log("open-url url", url)
+            const parsedUrl = new URL(url)
+            const target = new URL(`http://localhost:${this.serverPort()}`)
+            console.log("open-url search", parsedUrl.search)
+            target.search = parsedUrl.search // ?code=..&state=..
+            console.log("open-url target", target)
+            this.window?.webContents.send(ipc.channel.authenticatedRedirect, target.href)
```

Comment:
this code is complex and does not have explanation. The explanation can be either by comments or well, syntactically named variables.